### ISSUE TITLE
Add `deref` word into the vocabulary exclusions

### ISSUE THREAD

**Issue by SomeoneToIgnore** [intellij-rust/intellij-rust#4447]
## Environment

* **IntelliJ Rust plugin version:** 0.2.106.2134-193
* **Rust toolchain version:** 1.37.0 (eae3437df 2019-08-13) x86_64-apple-darwin
* **IDE name and version:** CLion 2019.3 EAP (CL-193.3519.24)
* **Operating system:** macOS 10.14.6

## Problem description

Consider the code:

```rust
struct Test {}

impl Deref for Test {
    type Target = ();

    fn deref(&self) -> &Self::Target {
        unimplemented!()
    }
}
```

Expected: no warnings or other issues

Actual: `deref` word is highlighted as typo:

![image](https://user-images.githubusercontent.com/2690773/65609992-59cc0f00-dfb9-11e9-8487-62983f65f479.png)

Since it's not a typo, it would be great not to highlight it.
